query-node/mappings/workingGroup.ts

```diff
@@ -0,0 +1,153 @@
+    return
+  }
+
+  const [, applicationIdToWorkerIdMap] = new StorageWorkingGroup.OpeningFilledEvent(event).params
+  const workerIds = [...applicationIdToWorkerIdMap.values()]
+
+  for (const workerId of workerIds) {
+    await createWorker(store, workerId, workerType, event)
+  }
+
+  // emit log event
+  logger.info('Workers have been created', { ids: workerIds.map((item) => item.toString()), workerType })
+}
+
+export async function workingGroup_WorkerStorageUpdated({ event, store }: EventContext & StoreContext): Promise<void> {
+  const workerType = getWorkerType(event)
+  if (!workerType) {
+    return
+  }
+  const [workerId, newMetadata] = new StorageWorkingGroup.WorkerStorageUpdatedEvent(event).params
+
+  // load worker
+  const worker = await store.get(Worker, {
+    where: {
+      workerId: workerId.toString(),
+      type: workerType,
+    },
+  })
+
+  // ensure worker exists
+  if (!worker) {
+    return inconsistentState('Non-existing worker update requested', workerId)
+  }
+
+  worker.metadata = bytesToString(newMetadata)
+
+  await store.save<Worker>(worker)
+
+  // emit log event
+  logger.info('Worker has been updated', { workerId, workerType })
+}
+
+export async function workingGroup_TerminatedWorker({ event, store }: EventContext & StoreContext): Promise<void> {
+  const workerType = getWorkerType(event)
+  if (!workerType) {
+    return
+  }
+  const [workerId] = new StorageWorkingGroup.TerminatedWorkerEvent(event).params
+
+  // do removal logic
+  await deactivateWorker(store, event, workerType, workerId)
+
+  // emit log event
+  logger.info('Worker has been removed (worker terminated)', { workerId, workerType })
+}
+
+export async function workingGroup_WorkerExited({ event, store }: EventContext & StoreContext): Promise<void> {
+  const workerType = getWorkerType(event)
+  if (!workerType) {
+    return
+  }
+  const [workerId] = new StorageWorkingGroup.WorkerExitedEvent(event).params
+
+  // do removal logic
+  await deactivateWorker(store, event, workerType, workerId)
+
+  // emit log event
+  logger.info('Worker has been removed (worker exited)', { workerId, workerType })
+}
+
+export async function workingGroup_TerminatedLeader({ event, store }: EventContext & StoreContext): Promise<void> {
+  const workerType = getWorkerType(event)
+  if (!workerType) {
+    return
+  }
+  const [workerId] = new StorageWorkingGroup.WorkerExitedEvent(event).params
+
+  // do removal logic
+  await deactivateWorker(store, event, workerType, workerId)
+
+  // emit log event
+  logger.info('Working group leader has been removed (worker exited)', { workerId, workerType })
+}
+
+/// ///////////////// Helpers ////////////////////////////////////////////////////
+
+function getWorkerType(event: SubstrateEvent): WorkerType | null {
+  if (event.section === 'storageWorkingGroup') {
+    return WorkerType.STORAGE
+  } else if (event.section === 'gatewayWorkingGroup') {
+    return WorkerType.GATEWAY
+  }
+  return null
+}
```

Comment:
A cosmetic tip. According to the so-called Object Calisthenics, it is usually a good idea to evade writing of `else`/`elseif` part of conditions and rather escape function inside `if` clause or create a helper function. 

We can simply remove `else` part in this particular function because `if` part escapes the function already.

```suggestion
  if (event.section === 'storageWorkingGroup') {
    return WorkerType.STORAGE
  } 
  
  if (event.section === 'gatewayWorkingGroup') {
    return WorkerType.GATEWAY
  }
  
  return null
}
```